core/storage/file/gae_models_test.py

```diff
@@ -0,0 +1,192 @@
+
+    def test_get_new_id_raises_not_implemented_error(self):
+        with self.assertRaises(NotImplementedError):
+            file_models.FileMetadataModel.get_new_id('entity1')
+
+    def test_get_undeleted_with_two_undeleted_models_returns_both(self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model1.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 2)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+        self.assertEqual(
+            file_metadata_model2, file_metadata_model_list[1])
+
+    def test_get_undeleted_with_one_deleted_and_undeleted_model_returns_one(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model1.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file2.png')
+        file_metadata_model2.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        file_metadata_model2.delete(
+            feconf.SYSTEM_COMMITTER_ID, 'Delete second file model')
+        file_metadata_model_list = (
+            file_models.FileMetadataModel.get_undeleted())
+        self.assertEqual(len(file_metadata_model_list), 1)
+        self.assertEqual(
+            file_metadata_model1, file_metadata_model_list[0])
+
+    def test_create_creates_model_with_correct_id_and_deleted_status_as_true(
+            self):
+        file_metadata_model1 = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        self.assertEqual(file_metadata_model1.deleted, False)
+        self.assertEqual(
+            file_metadata_model1.id, '/exp_id1/path/to/file1.png')
+
+        file_metadata_model2 = file_models.FileMetadataModel.create(
+            'exp_id1', '/exp_id1/path/to/file2.png')
+        self.assertEqual(file_metadata_model2.deleted, False)
+        self.assertEqual(
+            file_metadata_model2.id, '/exp_id1/path/to/file2.png')
+
+    def test_get_model_returns_the_correct_model_if_model_is_present(self):
+        file_metadata_model = file_models.FileMetadataModel.create(
+            'exp_id1', 'path/to/file1.png')
+        file_metadata_model.commit(feconf.SYSTEM_COMMITTER_ID, [])
+        actual_model = file_models.FileMetadataModel.get_model(
+            'exp_id1', 'path/to/file1.png')
+
+        self.assertEqual(actual_model, file_metadata_model)
+
+    def test_get_model_returns_None_if_model_is_absent_and_strict_fail_is_false(
```

Comment:
Nit: can just be 'none' rather than 'None' for consistency with other words in test names.